sonarqube/resource_sonarqube_group_member.go

```diff
@@ -0,0 +1,210 @@
+
+	// We need to check if a user is already a member in advance because SQ does not report this conflict in the add_user API call:
+	exists, _ := checkGroupMemberExists(d.Get("name").(string), d.Get("login_name").(string), m)
+	if exists {
+		return fmt.Errorf("resourceSonarqubeGroupMemberCreate: Group membership already exists: %+v", groupMembershipId)
+	}
+
+	resp, err := httpRequestHelper(
+		m.(*ProviderConfiguration).httpClient,
+		"POST",
+		sonarQubeURL.String(),
+		http.StatusNoContent,
+		"resourceSonarqubeGroupMemberCreate",
+	)
+	if err != nil {
+		return fmt.Errorf("error adding user to Sonarqube group: %+v", err)
+	}
+	defer resp.Body.Close()
+
+	d.SetId(groupMembershipId)
+
+	return resourceSonarqubeGroupMemberRead(d, m)
+}
+
+func resourceSonarqubeGroupMemberRead(d *schema.ResourceData, m interface{}) error {
+	sonarQubeURL := m.(*ProviderConfiguration).sonarQubeURL
+	sonarQubeURL.Path = strings.TrimSuffix(sonarQubeURL.Path, "/") + "/api/user_groups/users"
+	sonarQubeURL.RawQuery = url.Values{
+		"name": []string{d.Get("name").(string)},
+		"q":    []string{d.Get("login_name").(string)},
+	}.Encode()
+
+	resp, err := httpRequestHelper(
+		m.(*ProviderConfiguration).httpClient,
+		"GET",
+		sonarQubeURL.String(),
+		http.StatusOK,
+		"resourceSonarqubeGroupMemberRead",
+	)
+	if err != nil {
+		return fmt.Errorf("error reading Sonarqube group members: %+v", err)
+	}
+	defer resp.Body.Close()
+
+	readSuccess := false
+	// Decode response into struct
+	groupMemberReadResponse := GetGroupMembersResponse{}
+	err = json.NewDecoder(resp.Body).Decode(&groupMemberReadResponse)
+	if err != nil {
+		return fmt.Errorf("resourceSonarqubeGroupRead: Failed to decode json into struct: %+v", err)
+	}
+	// Loop over all returned members to see if the member we need exists.
+	for _, value := range groupMemberReadResponse.Members {
+		if d.Get("login_name").(string) == value.LoginName {
+			// If it does, set the values of that group membership
+			d.SetId(createGroupMembershipId(d.Get("name").(string), d.Get("login_name").(string)))
+			d.Set("name", d.Get("name").(string))
+			d.Set("login_name", value.LoginName)
+			readSuccess = true
+		}
+	}
+
+	if !readSuccess {
+		// Group member not found
+		d.SetId("")
+	}
+
+	return nil
+}
+
+func resourceSonarqubeGroupMemberDelete(d *schema.ResourceData, m interface{}) error {
+	sonarQubeURL := m.(*ProviderConfiguration).sonarQubeURL
+	sonarQubeURL.Path = strings.TrimSuffix(sonarQubeURL.Path, "/") + "/api/user_groups/remove_user"
+
+	sonarQubeURL.RawQuery = url.Values{
+		"name":  []string{d.Get("name").(string)},
+		"login": []string{d.Get("login_name").(string)},
+	}.Encode()
+
+	resp, err := httpRequestHelper(
+		m.(*ProviderConfiguration).httpClient,
+		"POST",
+		sonarQubeURL.String(),
+		http.StatusNoContent,
+		"resourceSonarqubeGroupMemberDelete",
+	)
+	if err != nil {
+		return fmt.Errorf("error deleting Sonarqube group member: %+v", err)
```

Comment:
adding more context to the error message and also wrap the original error:

```suggestion
		return fmt.Errorf("error deleting Sonarqube group member for group '%s' and user '%s': %w", d.Get("name").(string), d.Get("login_name").(string), err)
```